apiary.apib

```diff
@@ -71,6 +71,16 @@ All parameters and request attributes are optional by default.
 ### Get menu [GET /menu?date={date}&timespan={timespan}&firstDay={firstDay}&canteenId={canteenId}]
 Get menu for customer. If no attributes are provided menu is based on customer default preferences set in Septim.
 
+!! Podle mně je zanoření denní nabídky do subobjektu zbytečné (pole objektů které obsahují další pole objektů), udělal bych to v jedné úrovni jako pole objektů s tím, že atributem objektu by bylo datum. Takže by existovalo více objektů se stejným datem. Ale možná k Vašemu řešení máte nějaký důvod. Dalším důvodem proč mi to nedává smysl je to, že zanořujete datum, ale už nezanořujete jídlo (oběd, večeře).
+!! Přijde mi divné dávat pole na začátek a jednoduché atributy na konec. Jsem zvyklý to dělat obráceně.
+!! Informace o jídle máte neúplné. Chybí 
+  * Skupina jídel - Oběd, Večeře
+  * Filtr .. chybí hodnota podle které se bude filtrovat
+  * Alergeny .. Asi je najdete v informacích o recepturách, ale je otázkou, jestli je nutné volat pro každou položku další dotaz. Podle mně ne. Totéž platí i o názvu receptury. 
+!! Neexistuje žádné "ID of menu item". Existuje pouze ID jídla. Za mně by jste měl převést "jídlo" i "recept" na dvě položky, id záznamu a název záznamu. 
+!! Když jsem přidával "zdravost" k existujícím objednávkám tak jsem jí zapomněl přidat i sem. Respektive má být tady a ne v objednávkách. 
+!! Musím říct, že mně nebaví porovnávat jestli jste správně opsal hodnoty které jsem sepsal jako potřebné. Rád bych svojí práci napřel užitečnějším směrem. 
+
```

Comment:
Zanoření je otázka preference, náš navrhovaný způsob je elegantnější pro výpis. V druhém případě se převedou klíče (datumy) na pole a klíče se vypisují jednotlivé dny. Nicméně rozdíl v náročnosti je zanedbatelný a pokud vám více vyhovuje mít dny jako klíče, klidně to tak může být.
Skupina jídel a filtr jsou odlišné věci? Měl jsem za to, že se jedná o stejnou věc, v návrhu je filtr uveden jako oběd, večeře. Pokud jsou to jiné věci, tak skupiny jsou natvrdo definované jako Oběd a večeře nebo jsou to proměnné hodnoty?
Pořadí Vám přeskládám. 
V mé terminologii je MenuItem = jídlo (položka v nabídce), Product = receptura. Klidně vám to přejmenuji, stačí napsat co bude vyhovovat více.
Zdravost a alergeny budou zadány u jídla a u receptury nikoliv?

Reply:
- Poprosil bych práci s datem tak, aby se udržela datová logika napříč DB a klientem. Tedy že datum je jedním z parametrů objednávky. 
- Co se týče skupiny jídel a filtru, tak ve Vašem návrhu chybělo oboje. Ve valné většině se budou shodovat, ale rozhodně bych je řešil samostatně, tak jak to bylo v mém návrhu. 
- Skupiny jsou proměnné hodnoty
- Menu item a product jsou OK
- Zdravost a alergeny:
  - 1) Přece to nebudeme dělat tak, že část informací o receptuře se bude vracet v této funkci (název, popis, obrázek) a další část se bude dotahovat z receptury. Informace o receptuře budou buď součástí MenuItem nebo tam bude jenom odkaz na recepturu. Pokud máte receptury nacacheované, tak bych preferoval odkaz na recepturu. 
  - 2) Nevím jak alergeny, ale zdravost nyní není rozhodně součástí informací o receptuře. Takže se bude muset upravit API pro receptury.

Reply:
Zapracováno, zdravost a alergeny jsou součastí jídla, receptera se řeší jako odkaz a dotahuje pomocí GET Product details